torch/_functorch/_aot_autograd/logging_utils.py

```diff
@@ -65,7 +65,7 @@ def iter_graph(roots):
         seen = set()
         q = collections.deque()  # type: ignore[var-annotated]
         for node in roots:
-            if node is not None:
+            if node is not None and node not in seen:
```

Comment:
What was the problem?

Reply:
We only want to traverse over each node in the graph exactly once, and we do that by inserting nodes into the "seen" set. The issue is that we forget to check the "seen" set when inserting the root nodes. Typically that is not a problem, because the root nodes are from the different outputs and thus usually correspond to different nodes. With split_with_sizes, though all of the outputs correspond to the same node, ands this leads to the node being iterated over 3 times, and 3 sets of hooks being attached to the same node.

Reply:
I see, thank you for the explanation. Would be good to include in the commit message.